### PR TITLE
ci: migrate from Travis CI to GitHub Actions

### LINKED ISSUE #19: Convert to using GitHub actions for CI (rather than Travis CI).
This repository needs to be tested on Mac / Windows / Linux and on as many Linux variants as possible. This means it is probably better suited to GitHub actions.

### PR DESCRIPTION
**Pull request by umarcor**
Close #19

This PR migrates Linux, Windows and macOS jobs from Travis CI to a GitHub Actions workflow.

@mithro, these tests are native (Ubuntu 18.04, Windows Server 2019, and macOS). Which other GNU/Linux variants do you want to test (through containers)?

### CODE REVIEW

**Comment by mithro** [comment]
I would like to test as many popular Linux variants as we can.

File: .github/workflows/test.yml
Comment (mithro) [inline .github/workflows/test.yml:34]: Does git need to be told to enable symlinks on Windows like it does on Travis?

**Comment by umarcor** [comment]
> I would like to test as many popular Linux variants as we can.

I need some definition of popularity...

- Debian Buster, Sid
- Ubuntu 18.04, 19.04, 20.04
- Fedora 32, 33, 34
- CentOS 7, 8
- Arch Linux
- Alpine Linux 3.11, 3.12
- OpenSuse?

File: .github/workflows/test.yml
Comment (umarcor) [reply .github/workflows/test.yml:37]: I now added it to the previous step.

**Comment by mithro** [comment]
I would say;
 * Debian stable, Debian unstable, /maybe/ Debian testing
 * Currently supported LTS versions of Ubuntu + the latest version of Ubuntu (it not an LTS).
 * What ever is the equivalent of the above for Fedora / CentOS.
 * Your favorite Linux variant?

**Review by mithro — APPROVED** [review]
LGTM!

The workflow won't run until we merge this right?

**Comment by umarcor** [comment]
> The workflow won't run until we merge this right?

That's correct. Meanwhile, you can check the runs in my fork: https://github.com/umarcor/make-env/actions

Let's have this merged and I'll add the container based jobs later.

File: .github/workflows/test.yml
Comment (mithro) [reply .github/workflows/test.yml:37]: Any idea why this didn't fail on the windows github action previously?

Comment (umarcor) [reply .github/workflows/test.yml:37]: My understanding is that symlinks were traditionally not supported on Windows. However, some of the latest versions do support them. My guess is that the Windows version used in GHA is newer than the one in Travis. GHA is Azure, after all.

On the other hand, GHA has been using Git for Windows' bash as the default bash shell on Windows. However, that changed to MSYS2 some days ago, and it's probably going to be reverted, before switching again. See https://github.com/actions/virtual-environments/issues/1525 and this comment from Git for Windows' maintainer in MSYS2's gitter channel: [October 21, 2020 9:43 AM](https://gitter.im/msys2/msys2?at=5f8fe6bd7be0d67d278875d7).

So, I cannot really tell.

**Pull request merged**
